Utils.py

```diff
@@ -969,11 +969,8 @@ def __len__(self):
         return sum(len(iterable) for iterable in self.iterable)
 
 
-def is_iterable_of_str(obj: object) -> TypeGuard[typing.Iterable[str]]:
+def is_iterable_but_str(obj: object) -> TypeGuard[typing.Iterable[typing.Any]]:
     """ but not a `str` (because technically, `str` is `Iterable[str]`) """
```

Comment:
```suggestion
    """ `str` is `Iterable`, but that's not what we want """
```
docstring could also be changed to match the change in what it does